pkgdb/src/pkgdb/input.cc

```diff
@@ -124,7 +148,7 @@ PkgDbInput::scrapePrefix( const flox::AttrPath & prefix )
     std::make_tuple( prefix, static_cast<flox::Cursor>( root ), row ) );
 
   /* Start a transaction */
-  sqlite3pp::transaction txn( dbRW->db );
+  dbRW->db.execute( "BEGIN EXCLUSIVE TRANSACTION" );
```

Comment:
nonblocking: do we have any other writes? Why not retry this statement rather than the one in `connect()`?

Reply:
Looks like there are some more in our initialization logic - do we need to make those exclusive? Or we're just banking on the fact those won't conflict?

Reply:
I think this is explained by my comment above, we should already have an exclusive write lock on the database for all of these operations.